### PR TITLE
PREOPS-1152: Document plan for DP0.3

### PR DESCRIPTION
**Pull request by gpdf**
Please review this initial draft so we can get a version on `main`.  It will - of course - be further refined.

### CODE REVIEW

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
This technote describes the means by which the desired DP0.3 dataset will be served through the services of the Rubin Science Platform.
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
(simulated data) series of Data Previews:
```

Comment (leannep) [reply index.rst]: We have now defined all "Data Previews" to be based on LSST or LSST-like (simulated) data only.  Precursor datasets do not form part of the DP program

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
not tied to image data nor any output of the LSST Science Pipelines.
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
This "DP0.3" would provide some exposure to the Rubin Science Platform (RSP) and the planned Solar System data model to Data Preview delegates, and allow the early testing of certain features of the RSP that are specific to Solar System (or, more precisely, moving-object) data.
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
Data Preview 0.3 is planned for release to users sometime between June and September 2023. \citeds{rtn-011}
```

**Review by frossie — APPROVED** [review]

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
  For the most part this is not a concern, since those capabilities are currently being exercised by DP0.2 and will be exercised by DP1 and DP2 in future, but there is at least one relevant special case:
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
  - Support for time-series queries on moving objects in the RSP that appear together with appropriate cutouts
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
    in the images on which the points in the time series were observed, i.e., following the object across the sky, is planned however 
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
    this will not be possible in DP0.3.
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
    to the associated images will not be available.
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
- As the data are the result of a "fast simulation" without imaging, none of the capabilities of the RSP dealing with imaging data will be exercised.
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
- As the DP0.3 dataset is the result of a static simulation, rather than a progressive nightly release through the PPDB, as we will have in Operations,
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
  DP0.3 will not exercise certain dynamic aspects of the data model, such as the transformation from DIAObject to SSObject when
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
  It will therefore also not exercise the ability of users of the RSP to follow and/or replay that evolution.
```

File: index.rst
Comment (leannep) [inline index.rst:92]: I like this idea and support including it as a **stretch** goal, i.e. we would not hold up the release of DP0.3 if this functionality were not done (or complete). It is also something that could be easily added post-DP0.3 release as an additional capability. Could you add a statement making this clear.

File: index.rst
Comment (leannep) [inline index.rst:94]: Where would the visit images come from given that DP0.3 does not contain images.

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
The RSP instance for DP0.3 will be in the  Interim Data Facility (IDF) hosted on the Google cloud platform, and, we argue below,
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
The catalogs will be held in a PostgreSQL database, most likely on-premises at the USDF,
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
Qserv is not a suitable choice for the database back end, for two reasons:
```
```suggestion
Qserv is probably not a suitable choice for the database back end, for two reasons:
```

File: index.rst
Comment (leannep) [inline index.rst:130]:
```suggestion
In the "Hybrid Model" for the US DAC, planned from October 2023, the user-facing services will be in the Google cloud, with the data back ends at the USDF.
```
```suggestion
In the "Hybrid Model" for the US DAC, the user-facing services will be in the Google cloud, with the data back ends at the USDF.
```

File: index.rst
Comment (leannep) [inline index.rst:158]: Suggest adding a note clarifying that deploying a SSO specific TAP service is an interim work around and that in the eventual operations-era services there will be one TAP service for all LSST catalogs

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
The RSP Scientist, Gregory Dubois-Felsmann,  would develop this Felis, based on existing DPDD Felis code,
```

File: index.rst
Comment (leannep) [inline index.rst]: Suggest to remove "a few hours" and replace with "can be easily written and released on the timescale for DP0.3" - we have a very bad track record of underestimating how long things take - the key point is that this can be easily done in the timeframe for DP0.3.

File: index.rst
Comment (leannep) [inline index.rst:225]: Is there any possibility for users to specify their own default and have that saved as a user setting that would be default every time the user logs in?  The groups of users interested in each of DP0.2 and DP0.3 are very distinct.

File: index.rst
Comment (leannep) [inline index.rst:228]: suggest to add "and hence the optimal order can be set in agreement with the SSSC"

File: index.rst
Comment (leannep) [inline index.rst:241]: This is good but as it requires development, if that development is delayed, it should not delay the release of DP0.3, i.e if we delivered this functionality a few months after the main release as an improvement, that would be preferable to delaying DP0.3 . Could you add a comment to that effect.

File: index.rst
Comment (leannep) [inline index.rst:264]: Could you not just write another convenience function specific to the solar system TAP service that is different  from the existing one as interim workaround, e.g "get_tap_service()" and "get_sso_tap_service", with the latter pointing to the TAP URL for the SSO catalogs?

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
 Develop tutorials and notebooks (Community Engagement Team (CET))
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
- 26 May 2023: Release candidate turned over to product owners (Dubois-Felsmann & Slater)
```

File: index.rst
Comment (leannep) [inline index.rst]:
```suggestion
The SSSC have stated that they want to  regenerate the simulation based on the recently released V3.0 of the baseline survey.
```

Comment (leannep) [reply index.rst]: and add a citation to PSTN-055 - SCOC phase two survey recommendations

File: index.rst
Comment (leannep) [inline index.rst]: Which notebook? Replace this statement with a link to the notebook (or other documentation) where this is described

File: index.rst
Comment (leannep) [inline index.rst:336]: Missing a closing end '"'

File: index.rst
Comment (leannep) [inline index.rst:344]: Can the above text be reformulated into a coherent statement as to what we will do (or the options for what we might do) rather that a quote from a Jira ticket.  Keep a reference to he Jira ticket though.

File: index.rst
Comment (leannep) [inline index.rst:360]: What is the ETA for filling these section in? We could release a first version now without this but if so, add a line that these sections will be completed once we have a better idea of the implementation details

**Review by leannep — CHANGES_REQUESTED** [review]

**Comment by leannep** [comment]
Nice work. as an overall comment, it might be good to state clearly what is needed for a DP0.3 release and what is a stretch goal that could be added a short time after a DP0.3 release, e.g. include a simple visit table, perhaps in ObsCore format.

File: index.rst
Comment (gpdf) [reply index.rst:336]: It's not - that's conventional American copy-editing for a multi-paragraph quote.

https://www.grammar-monster.com/punctuation/quotation_marks_multiple_or_new_paragraphs.html

File: index.rst
Comment (gpdf) [reply index.rst]: Good catch - I wrote this before the final decision about DP1.

File: index.rst
Comment (gpdf) [reply index.rst]: Thanks - again, this was drafted before that decision.

**Comment by gpdf** [comment]
Will process comments this weekend.  Thanks!

File: index.rst
Comment (ctslater) [inline index.rst]: I'm not sure what reconstructed means. I think this is more like "Measured orbital parameters for detected Solar System objects"

File: index.rst
Comment (ctslater) [inline index.rst]: "Observation-specific properties of a detected Solar System object"?

File: index.rst
Comment (ctslater) [inline index.rst]:
```suggestion
The presentation order of tables, and of columns within tables, are controlled by the Felis-based
```

File: index.rst
Comment (ctslater) [reply index.rst:344]: I would drop this ingest section because these are implementation details.

File: index.rst
Comment (ctslater) [reply index.rst:92]: I'm a bit wary of dangling a nice-sounding feature if there's nobody who is well-positioned to implement it. Opsim -> ObsCore/Visit table is not a production data path and I'm not aware of any DM code to do that, so it would be "new scope" just for DP0.3 for someone in DM to implement that.

File: index.rst
Comment (ctslater) [reply index.rst]: I'm not really sure what this means.

File: index.rst
Comment (ctslater) [inline index.rst]:
```suggestion
  - During full operations with LSSTCam, the RSP will support time-series queries for moving objects together with appropriate cutouts
```

File: index.rst
Comment (gpdf) [reply index.rst:344]: The intent was to include implementation details in the technote as the work progresses.

File: index.rst
Comment (gpdf) [reply index.rst]: This might be an HEP-ism in my vocabulary.  Just to explain myself: I tend to use "detect" for something more "done in one place" - like "detected source on an image", and "reconstruct" for "concluded from combining several pieces of evidence".  E.g., "the transient was first detected on February 1st; by April it had been reconstructed as an SNIa candidate and its redshift measured".

But I'm happy to make the change you suggested.

File: index.rst
Comment (gpdf) [reply index.rst:92]: @ctslater A few points:

- The only even remotely non-trivial thing about the conversion from OpSim to ObsCore is the computation of the visit outline on the sky from the boresight and rotation.  But this conversion must already have been done in order to execute the SSO simulation, so it seems like code should be available.
- Starting from that point, I'd expect to do the table conversion myself, for this static dataset.
- It is, in fact, an LSST requirement to produce and serve an ObsCore-like table based on scheduler outputs - this is the ObsLocTAP service.  I'll make this explicit in the document.
- The ability to display projected visits as outlines on the sky is not in the DM requirements, but it would be quite odd for us not to provide users with a way to visualize the output of the ObsLocTAP service, and in fact the Portal will do this for free when presented with the resulting table, or a subset of rows from it.

File: index.rst
Comment (gpdf) [reply index.rst:94]: They are just the polygonal outlines of the visits (as a whole, not broken down into detectors) on the sky.

File: index.rst
Comment (gpdf) [reply index.rst:130]: I'm not sure I understand the date reference.  Do we have a specific milestone for this?  I don't see one in DMTN-158 or DMTN-232.

File: index.rst
Comment (gpdf) [reply index.rst]: Addressed, though a bit under protest, because the ease of doing this is a pretty significant feature of the RSP architecture.

File: index.rst
Comment (gpdf) [reply index.rst:225]: Not on the DP0.3 time scale, no.

However, I expect we can provide DP0.3 users a DP0.3-specific URL, which they could bookmark, for instance, for starting the Portal already pointed to the DP0.3 TAP service.

File: index.rst
Comment (gpdf) [reply index.rst:264]: We could, but I'd view that as a Plan B fallback, too trivial to mention here.  We need the generalization for other reasons as well.

File: index.rst
Comment (gpdf) [reply index.rst]: Rewritten a bit differently.

File: index.rst
Comment (gpdf) [reply index.rst]: It means that at this point we sign off (or not!) on it being acceptable for release to the Delegates.

File: index.rst
Comment (leannep) [reply index.rst:130]: It is the date when SLAC takes over the google contract, which is being taken as the transition point to the hybrid model. DP0 should refer to 'IDF' DP1/DP2 will be in the google cloud-hosted IDF

**Review by leannep — APPROVED** [review]

File: index.rst
Comment (ctslater) [reply index.rst:92]: I agree it is not a huge amount of work, but it is also not zero work, so we should identify who is responsible for providing it. Just trying to avoid a situation where someone gets volunteered at the last minute.

File: index.rst
Comment (ctslater) [reply index.rst]: Ah right, if I heard "reconstructed particle track" I would probably have had the right association, but I think it's an unfamiliar phrasing in astronomy.

File: index.rst
Comment (ctslater) [reply index.rst:344]: I would be ok with reporting on the implementation once it's decided, but I'd like DAX to be able to have a look and consider the options before we start publicizing that.

**Comment by gpdf** [comment]
I'm merging at this point, with Leanne and Frossie's approvals, and we will, as planned, continue to refine the document as we go on into implementation.

**Pull request merged**
